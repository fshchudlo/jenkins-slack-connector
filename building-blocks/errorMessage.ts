import { section } from "./section";

export function errorMessage(message: string, details: any) {
    return [
        section(`:octagonal_sign: *${message}* \n\`${details}\``)
    ];
}