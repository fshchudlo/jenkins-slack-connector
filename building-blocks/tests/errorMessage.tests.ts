import { errorMessage } from "../errorMessage";

describe("errorMessage", () => {
    it("should create an error message section", () => {
        const control = errorMessage("Something went wrong", "Error details");
        expect(control).toMatchSnapshot();
    });
});
