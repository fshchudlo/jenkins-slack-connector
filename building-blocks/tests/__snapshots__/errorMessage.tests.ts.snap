// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorMessage should create an error message section 1`] = `
[
  {
    "text": {
      "text": ":octagonal_sign: *Something went wrong* 
\`Error details\`",
      "type": "mrkdwn",
    },
    "type": "section",
  },
]
`;
